tests/beta/test_default_tensor.py

```diff
@@ -331,6 +338,298 @@ def test_hermitian_exceptions(self):
             hermitian(H2)
  
 
+class TestDefaultTensorNetwork:
+    """Tests of the basic tensor network functionality of default.tensor plugin."""
+
+    def clear_network(self, tensornet_device_2_wires):
+        """Tests that the _clear_network method clears the relevant bookkeeping data."""
+
+        dev = tensornet_device_2_wires
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.PauliX(wires=0)
+            qml.PauliY(wires=1)
+            return qml.expval(qml.PauliZ(0)), qml.sample(qml.PauliY(1))
+
+        circuit()
+        dev.reset()
+
+        assert dev._nodes == {}
+        assert not dev._contracted
+        assert dev._terminal_edges == []
+
+
+    def test_reset(self, tensornet_device_2_wires, tol):
+        """Tests that the `reset` method clears relevant bookkeeping data and re-initializes the iniital state."""
+
+        dev = tensornet_device_2_wires
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.PauliX(wires=0)
+            qml.PauliY(wires=1)
+            return qml.expval(qml.PauliZ(0)), qml.sample(qml.PauliY(1))
+
+        circuit()
+        dev.reset()
+
+        assert 'state' in dev._nodes and len(dev._nodes) == 1
+        assert len(dev._nodes['state']) == 2
+        assert all([dev._nodes['state'][idx].name == "ZeroState({},)".format(idx) for idx in range(2)])
+        assert np.allclose([dev._nodes['state'][idx].tensor for idx in range(2)], dev.zero_state, atol=tol, rtol=0)
+        assert not dev._contracted
+        assert len(dev._terminal_edges) == 2
+        node_edges = [dev._nodes['state'][idx].edges for idx in range(2)]
+        node_edges_set = set([edge for sublist in node_edges for edge in sublist])
+        assert node_edges_set == set(dev._terminal_edges)
+
+
+    def test_add_initial_state_nodes_2_wires(self, tensornet_device_2_wires):
+        """Tests that the initial states are properly created for a 2 wire device."""
+
+        dev = tensornet_device_2_wires
+        dev._clear_network()
+
+        # factorized state
+        tensors = [np.array([1., 0.]), np.array([np.sqrt(0.5), -1j * np.sqrt(0.5)])]
+        wires = [[0], [1]]
+        names = ["AliceState", "BobState"]
+        dev._add_initial_state_nodes(tensors, wires, names)
+
+        assert "state" in dev._nodes and len(dev._nodes) == 1
+        assert len(dev._nodes["state"]) == 2
+        assert dev._nodes["state"][0].name == "AliceState(0,)"
+        assert dev._nodes["state"][1].name == "BobState(1,)"
+        assert edges_valid(dev, num_nodes=2)
+
+        # entangled state
```

Comment:
I was thinking to suggest that these two tests could be done using `@pytest.mark.parametrize`, but the assert statements are a bit variable.

It might make sense to split up into separate tests though, since we are testing two different things. This should also make it easier to debug failing tests. The same applies to the `test_add_initial_state_nodes_3_wires` test.

Reply:
Done!